### PR TITLE
fix: fixme handling with falsy values

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: src/workerRunner.ts [hunk 1]
```diff
@@ -484,7 +484,10 @@ function buildTestEndPayload(testId: string, testInfo: TestInfo): TestEndPayload
 
 const kTestArgsSymbol = Symbol('testargs');
 
-function modifier(testInfo: TestInfo, type: 'skip' | 'fail' | 'fixme' | 'slow', arg?: boolean | string | Function, description?: string) {
+function modifier(testInfo: TestInfo, type: 'skip' | 'fail' | 'fixme' | 'slow', args: [arg?: boolean | string | Function, description?: string]) {
+  if (args.length >= 1 && !args[0])
```
Comment (dgozman) [inline src/workerRunner.ts:488]: I'd prefer to rewrite the whole section now, like `if (args.length === 0) {} else if (args.length === 1) {} else {}`

File: test/test-modifiers.spec.ts [hunk 2]
```diff
@@ -100,6 +103,22 @@ test('test modifiers should work', async ({ runInlineTest }) => {
         test.skip(({ foo }) => !foo, 'reason');
         test('suite4', () => {});
       });
+
+      test('fixme1', async () => {
+        test.fixme(undefined)
```
Comment (dgozman) [inline test/test-modifiers.spec.ts:108]: style: missing semicolons

File: test/test-modifiers.spec.ts [hunk 3]
```diff
@@ -55,6 +55,9 @@ test('test modifiers should work', async ({ runInlineTest }) => {
       test('skipped6', async ({foo}) => {
         test.skip(({ foo }) => foo, 'reason');
       });
+      test('skipped7', async ({foo}) => {
```
Comment (dgozman) [inline test/test-modifiers.spec.ts:58]: Move non-skipped ones to the "passed" section instead.

File: test/test-modifiers.spec.ts [hunk 4]
```diff
@@ -100,6 +103,22 @@ test('test modifiers should work', async ({ runInlineTest }) => {
         test.skip(({ foo }) => !foo, 'reason');
         test('suite4', () => {});
       });
+
+      test('fixme1', async () => {
+        test.fixme(undefined)
+      });
+
+      test('fixme2', async () => {
+        test.fixme(null)
```
Comment (dgozman) [inline test/test-modifiers.spec.ts:112]: Let's instead add `fixme(undefined, 'reason')`.

File: .gitignore [hunk 5]
```diff
@@ -3,3 +3,4 @@ package-lock.json
 .vscode
 out/
 /test-results
+/tsconfig.tsbuildinfo
```
[the remark this answers is not in the document]
Comment (mxschmitt) [reply .gitignore:6]: For some reason this file was generated on my environment when I executed npm run build.

File: src/testType.ts [hunk 6]
```diff
@@ -103,16 +103,17 @@ export class TestTypeImpl {
     suite._hooks.push({ type: name, fn, location: callLocation() });
   }
 
-  private _modifier(type: 'skip' | 'fail' | 'fixme' | 'slow', arg?: boolean | string | Function, description?: string) {
+  private _modifier(type: 'skip' | 'fail' | 'fixme' | 'slow', ...modiferAgs: [arg?: boolean | string | Function, description?: string]) {
```
Comment (dgozman) [inline src/testType.ts:106]: `arg` cannot be a string now, right?

File: src/workerRunner.ts [hunk 7]
```diff
@@ -221,10 +221,10 @@ export class WorkerRunner extends EventEmitter {
         const basePath = path.join(this._project.config.snapshotDir, relativeTestFilePathForSnapshots, sanitizedTitle, testInfo.snapshotPathSegment);
         return path.join(basePath, ...pathSegments);
       },
-      skip: (arg?: boolean | string, description?: string) => modifier(testInfo, 'skip', arg, description),
-      fixme: (arg?: boolean | string, description?: string) => modifier(testInfo, 'fixme', arg, description),
-      fail: (arg?: boolean | string, description?: string) => modifier(testInfo, 'fail', arg, description),
-      slow: (arg?: boolean | string, description?: string) => modifier(testInfo, 'slow', arg, description),
+      skip: (...args: [arg?: boolean | string, description?: string]) => modifier(testInfo, 'skip', args),
```
Comment (dgozman) [inline src/workerRunner.ts:224]: Same here, `arg` cannot be a string.

File: src/workerRunner.ts [hunk 8]
```diff
@@ -414,17 +414,11 @@ function buildTestEndPayload(testId: string, testInfo: TestInfo): TestEndPayload
...
-    description = arg;
-  } else {
-    if (!arg)
-      return;
-  }
+function modifier(testInfo: TestInfo, type: 'skip' | 'fail' | 'fixme' | 'slow', modifierArgs: [arg?: boolean | string, description?: string]) {
+  if (modifierArgs.length >= 1 && !modifierArgs[0])
+    return;
 
+  const [arg, description] = modifierArgs;
```
Comment (dgozman) [inline src/workerRunner.ts:421]: `arg` is not used anymore.

File: src/workerRunner.ts [hunk 9]
```diff
@@ -414,17 +414,11 @@ function buildTestEndPayload(testId: string, testInfo: TestInfo): TestEndPayload
...
-  if (arg === undefined && description === undefined) {
-    // No parameters - modifier applies.
-  } else if (typeof arg === 'string') {
-    // No condition - modifier applies.
-    description = arg;
-  } else {
-    if (!arg)
-      return;
-  }
+function modifier(testInfo: TestInfo, type: 'skip' | 'fail' | 'fixme' | 'slow', modifierArgs: [arg?: boolean | string, description?: string]) {
```
Comment (dgozman) [inline src/workerRunner.ts:417]: `arg` cannot be a string.

File: src/testType.ts (hunk 6 above)
Comment (mxschmitt) [reply src/testType.ts:106]: by definition any would probably be the right type, since it needs to be something truehy. wdyt?

**Review by dgozman — APPROVED** [review]

**Pull request merged**